### DISCUSSION TITLE
Anyone else prefer to move off of GitHub?

### DISCUSSION THREAD

**Discussion by QuintillusCFC** [C7-Game/Prototype#451] [General]
With the news that GitHub is ingesting all non-private repos to train Copilot, my inclination to contribute to projects hosted on GitHub has declined considerably.  I'm curious if anyone else is in the same boat?  As a firm believer in human intelligence, supporting Microsoft's efforts to supplant that just does not site right with me.

All of my solo projects (including my Civ3 editor) have been on SourceHut (https://sourcehut.org) for many years, as most of them use Mercurial, which GitHub does not support.  It has an issue tracker, but its major downfall for multi-contributor projects is a lack of a "pull requests" equivalent.  Thus it's a bit hard to recommend as a realistic alternative for a project like this.

However, there are other options.  Codeberg looks promising and does have pull requests, issue tracker, wiki... all the major features (see https://codeberg.org/Codeberg/Documentation/pulls for a sample of how PRs work).  It may be the best human-intelligence-friendly alternative.

Heptapod is another option (https://heptapod.net/heptapod/), based on GitLab but hosted in the more privacy-friendly EU, and it offers free hosting for open source projects.

GitLab itself may also work although I'd want to review their AI policies first.

I understand that some developers like AI, so I don't expect this to be a universal view.  But the source of hosting location will be one of the factors I'm weighing when deciding if I will come back and contribute to Dutch features, or focus on other projects that are hosted on sites that have friendlier policies.

**Comment by pcen** [comment]
I don't mind where it's hosted / open to changing but preferably somewhere that supports git. I would be hesitant to move to GitLab because of this whole fiasco though: https://www.theregister.com/2022/08/05/gitlab_reverses_deletion_policy/

**Comment by WildWeazel** [comment]
I intentionally ignored this until Carthage was out the door, so I guess now I should respond.

Honestly, no. I am kind of an idealist and have no particular love for Microsoft, but for better or worse GitHub is the Hub of open source collaboration. I feel that it would only be a net loss to move to a less popular platform. We're not big enough to pull a following along with us, or necessarily even get people past the friction of setting up on a new relatively obscure site. This is, for the present, where the developers are. Personally I'm not particularly bothered by Copilot. Philosophical concerns with AI aside, with our code being MIT licensed there's nothing to deter anyone from feeding it to an AI anyway, nor from forking it back to GitHub (which has already been done) where it will be.

**Comment by TomWerner** [comment]
> as most of them use Mercurial, which GitHub does not support

I'll just note that [JJ](https://github.com/jj-vcs/jj) is a really nice frontend to git that gives me most of the mercurial features I'm used to (revsets, a rational set command line commands, commit visualization with reasonable defaults, commit splitting, etc). I've found it extremely useful when doing longer chains of commits.
